x-pack/plugins/ml/public/application/datavisualizer/index_based/components/search_panel/search_panel.tsx

```diff
@@ -4,37 +4,32 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { FC } from 'react';
+import React, { FC, useState } from 'react';
 
-import {
-  EuiFieldSearch,
-  EuiFlexItem,
-  EuiFlexGroup,
-  EuiForm,
-  EuiFormRow,
-  EuiIconTip,
-  EuiSuperSelect,
-  EuiText,
-} from '@elastic/eui';
+import { EuiFlexItem, EuiFlexGroup, EuiIconTip, EuiSuperSelect, EuiText } from '@elastic/eui';
 
 import { FormattedMessage } from '@kbn/i18n/react';
 import { i18n } from '@kbn/i18n';
 
 import { IndexPattern } from '../../../../../../../../../src/plugins/data/public';
 
 import { SEARCH_QUERY_LANGUAGE } from '../../../../../../common/constants/search';
-import { SavedSearchQuery } from '../../../../contexts/ml';
 
-// @ts-ignore
-import { KqlFilterBar } from '../../../../components/kql_filter_bar/index';
+import {
+  esKuery,
+  esQuery,
+  Query,
+  QueryStringInput,
+} from '../../../../../../../../../src/plugins/data/public';
 
 interface Props {
   indexPattern: IndexPattern;
-  searchString: string | SavedSearchQuery;
-  setSearchString(s: string): void;
-  searchQuery: string | SavedSearchQuery;
-  setSearchQuery(q: string | SavedSearchQuery): void;
+  searchString: Query['query'];
+  setSearchString(s: Query['query']): void;
+  searchQuery: Query['query'];
+  setSearchQuery(q: Query['query']): void;
   searchQueryLanguage: SEARCH_QUERY_LANGUAGE;
+  setSearchQueryLanguage(q: any): void;
```

Comment:
probably you can use `SEARCH_QUERY_LANGUAGE` from `x-pack/plugins/ml/common/constants/search.ts`